### PR TITLE
Grant TUF SA access to modify storage objects

### PR DESCRIPTION
**Pull request by haydentherapper**
Needed to use the TUF SA on GitHub Actions workflows.

### CODE REVIEW

**Review by jku — APPROVED** [review]
Looks correct.

For completeness (assuming we don't try to find another solution) we're also going to need `roles/compute.loadBalancerAdmin` or more specifically `compute.urlMaps.invalidateCache` permission but that is not really related to the bucket so this change seems good to me.

File: terraform/gcp/modules/tuf/tuf.tf [hunk 1]
```diff
@@ -119,3 +127,11 @@ resource "google_storage_bucket_iam_member" "public_tuf_preprod_member" {
...
   depends_on = [google_storage_bucket.tuf_preprod]
 }
+
+resource "google_storage_bucket_iam_member" "tuf_sa_preprod_editor" {
+  bucket = google_storage_bucket.tuf_preprod.name
+  role   = "roles/storage.objectUser"
+  member = format("serviceAccount:%s@%s.iam.gserviceaccount.com", var.tuf_service_account_name, var.project_id)
+
+  depends_on = [google_storage_bucket.tuf_preprod, google_service_account.tuf-sa]
+}
```
Comment (cpanato) [inline terraform/gcp/modules/tuf/tuf.tf:137]: nit: new line

**Review by cpanato — APPROVED** [review]
lgtm

**Comment by haydentherapper** [comment]
> For completeness (assuming we don't try to find another solution) we're also going to need roles/compute.loadBalancerAdmin or more specifically compute.urlMaps.invalidateCache permission but that is not really related to the bucket so this change seems good to me.

Planning to grant that in the private repo since the load balancer configuration isn't in scaffolding.

**Pull request merged**